lib/utils/deprecation/deprecator.rb

```diff
@@ -0,0 +1,108 @@
+require 'utils/deprecation/config_file'
+require 'inspec/log'
+
+module Inspec
+  module Deprecation
+    class Deprecator
+      attr_reader :config, :groups
+
+      # This is used only in functional testing
+      def self.class_test_cfg_io(io)
+        @@test_cfg_io = io # rubocop: disable Style/ClassVars
+      end
+
+      def initialize(opts = {})
+        @@test_cfg_io ||= nil # rubocop: disable Style/ClassVars
+        @config = Inspec::Deprecation::ConfigFile.new(opts[:config_io] || @@test_cfg_io)
+        @groups = @config.groups
+      end
+
+      def handle_deprecation(group_name, message, opts = {})
+        group = groups[group_name.to_sym] # Will be nil if unrecognized group
+        annotate_stack_information(opts)
+        assembled_message = assemble_message(message, group, group_name, opts)
+
+        action = (group ? group[:action] : :warn) || :warn
```

Comment:
This right here is a bug - if there is no group, we're supposed to use the unknown_group_action.  Instead, this always warns.